Impossible-Agent-Suite/staxmigrate/src/agent.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  Finding,
+  FindingType,
+  FindingSeverity,
+  TransactionEvent,
+  HandleTransaction
+} from 'forta-agent';
+
+let IF_ADDR = '0xb0e1fc65c1a741b4662b813eb787d369b8614af1';
+let ZERO_ADDR = '0x0000000000000000000000000000000000000000';
+
+export const IF_ABI = [
+  'event Transfer(address indexed from, address indexed to, uint256 value)',
+  'function staxMigrate(uint amount)'
+];
+
+export const provideHandleTransaction = (address: string): HandleTransaction => {
+  return async (tx: TransactionEvent): Promise<Finding[]> => {
+    // Setup the findings array
+    const findings: Finding[] = [];
+
+    // Get all times `Transfer` was emitted
+    const transfers = tx.filterLog(IF_ABI[0], address);
+
+    // Get all times `staxMigrate` was called
+    const migrations = tx.filterFunction(IF_ABI[1], address);
+
```

Comment:
You can switch between the to declarations and add in between something like `if (!migrations) return findings;` to avoid filtering unneeded Transfers when there is no `staxMigrate` call in the transaction

Reply:
Good idea, that has been implemented.